### PR TITLE
Doors

### PR DESCRIPTION
**Pull request by fsandel**
- [ ] implemented doors in parser/raycaster and collision detector
- [ ] added door toggle as keyhook, right now bound to space which toggles all surrounding doors in changeable range

### CODE REVIEW

File: src/doors.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	window = (t_window *)arg;
+	if (keydata.key == MLX_KEY_SPACE && keydata.action == MLX_PRESS)
+	{
+		toggle_surrounding_doors(window);
+		draw_scene(window);
+	}
+}
+
+static void	toggle_surrounding_doors(t_window *window)
```
Comment (philippschwarzmueller) [inline src/doors.c:17]: this segfaults if you are close to the maps border, because it cant iterate over the empty 'outside' cubes

**Review by philippschwarzmueller — CHANGES_REQUESTED** [review]
if you change the approach of checking the proximity of a door to casting a single centered ray instead of checking all fields around we get no segfaults and also dont open doors in the back of the players view

**Comment by fsandel** [comment]
uwu pls notice me senpai

**Review by philippschwarzmueller — APPROVED** [review]
lgtm

**Pull request merged**